ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -657,16 +649,23 @@ private void executeDdlCommand(DdlCommand ddlCommand) {
     }
   }
 
-  private Pair<AbstractStreamCreateStatement, String> maybeAddFieldsFromSchemaRegistry(
-      AbstractStreamCreateStatement streamCreateStatement,
+  private Pair<Statement, String> maybeAddFieldsFromSchemaRegistry(
+      Statement statement,
+      String statementText,
       Map<String, Object> streamsProperties
   ) {
-    Pair<AbstractStreamCreateStatement, String> avroCheckResult =
-        new AvroUtil().checkAndSetAvroSchema(
-            streamCreateStatement,
-            streamsProperties,
-            ksqlEngine.getSchemaRegistryClient()
-        );
-    return avroCheckResult;
+    if (statement instanceof AbstractStreamCreateStatement) {
+      AbstractStreamCreateStatement streamCreateStatement = (AbstractStreamCreateStatement)
+          statement;
+      Pair<AbstractStreamCreateStatement, String> avroCheckResult =
+          new AvroUtil().checkAndSetAvroSchema(
+              streamCreateStatement,
+              streamsProperties,
+              ksqlEngine.getSchemaRegistryClient());
+      if (avroCheckResult.getRight() != null) {
+        return new Pair<>(avroCheckResult.getLeft(), avroCheckResult.getRight());
```

Comment:
Also, i know this was here before, but not a fan of returning a `Pair` it doesn't really tell us anything. I'd rather have a specific class that represents the result. I put this in the same bucket as passing around/returning collection types etc everywhere when they would be better described as something that makes sense.

Reply:
I thought about this. I wasn't terribly bothered in this case since the usage is very local and the returned pair is immediately decomposed. I don't feel strongly about not doing it though. I'll update with a specific return type class.